Making SinglePhaseFluidProperties follow the naming conventions

andrsd opened the pull request:
This is part 2 of #12050 

Fixing all calls in the repo to use the right method name.

andrsd committed:
Fixed e(p, T)

Replacing e(p, T) and e_dpT(...) with e_from_p_T to following the naming
conventions.

Refs #12050

andrsd committed:
Fixed c(p, T)

Replacing c(p, T) with c_from_p_T(...) to follow naming conventions

Refs #12050

andrsd committed:
Fixing mu(p, T)

Replacing mu(p, T) and mu_dpT with mu_from_p_T(...).

Refs #12050

andrsd committed:
Fixing up h(p, T)

Redirecting the call to h_from_p_T()

andrsd committed:
Fixing beta(p, T)

Redirecting beta(p, T) to beta_from_p_T(p, T)

andrsd committed:
Fixing s(p, T)

Redirecting s(p, T) to s_from_p_T(p, T)

andrsd committed:
Fixing k(p, T)

k(p, T) replaced by k_from_p_T(...)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/229193/) on 6e3f49e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/12107/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/12107/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 720e3fcc7a60fa90356a7dfb01540a4db6b4ebb4`

andrsd committed:
clang-format

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit bd111f4421ee3145f7dde6ac14aa82966cd28653
+Subproject commit 4b120c7af7a95d45e4640afb2688cdba946a79c6
```
**Caution!** This contains a submodule update

andrsd replied:
Because I am sloppy with `git submodule update`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/229222/) on 393bfb0 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12107/site)

This comment will be updated on new commits.

andrsd committed:
Working around gcc-7.3.0 brain damage

While clang is happy compiling with just one overloaded method, gcc for
some reason would not compile it and would throw a warning that our
method is hidden even though it is not *sigh*.

andrsd commented:
Correct
